### DISCUSSION TITLE
does rustdesk can change relay/ id server address  by regedit or bat?

### DISCUSSION THREAD

**Discussion by acer1204** [rustdesk/rustdesk#3439] [General]
does rustdesk can change relay/ id server address  by regedit or bat?

**Comment by 0x100001** [comment]
No. You can edit that file: C:\Windows\ServiceProfiles\LocalService\AppData\Roaming\RustDesk\config\RustDesk2.toml

I am doing the same with my software to apply a custom configuration network wide.

**Reply by acer1204** [reply]
I change it , but when i restart rustdesk, this file context will recover by rustdisk.

**Comment by learningtime2025** [comment]
i think the better way is to change them in code, then compile binaries.
if you want to change from one server to another - remove rustdesk dir from user profile C:\Windows\ServiceProfiles\LocalService\AppData\Roaming\RustDesk then run new binary

**Comment by yashpatel1598** [comment]
i have my own server and i want to do hard coded my server ip and key so where i need to change in code for that i am on linux machine can you tell me

**Reply by 0x100001** [reply]
https://rustdesk.com/docs/en/self-host/hardcode-settings/

**Reply by yash1597** [reply]
@0x100001 but this is from github right? From setting action secrets i want to change in code so when i make build it comes with that my own server and key

**Reply by jacotec** [reply]
/libs/hbb_common/src/config.rs from line 80 are the coded servers and key

**Reply by latiaoge** [reply]
> /libs/hbb_common/src/config。rs行80是服务器编码和钥匙

thank you!

**Comment by korneevigor** [comment]
Работаю с 2 серверами ретрансляции.
написал скрипт на AHK для запуска Rustdesk под каждый сервер.
описание:
запускаю скрипт, он проверяет наличие \Rouming\RustDesk. Удаляет если каталог существует.
 распаковывает архив в папку  \Rouming\RustDesk  настроики и клиенты.
 при закрытии RustDesk происходит архивация  \Rouming\RustDesk  и удаление.

**Reply by korneevigor** [reply]
RustDesk,ahk
/*
Для программы RustDesk Portable
Сделано для того чтобы можно было запускать RustDesk с разными настройками.
Это дает возможность подключатся к разным серверам с сохранеными точками подключения.

*/


ArcPath = %A_WorkingDir%\rd.zip ;
IfExist, %A_AppData%\RustDesk 
	{
	FileRemoveDir, %A_AppData%\RustDesk, 1 ; Удаляем папку ...\Roaming\RustDesk 
	}
;распаковывается файл rd.zip конфигурации RustDesk в ...\Roaming\RustDesk
Shell := ComObjCreate("Shell.Application")
Items := Shell.NameSpace(ArcPath).Items
Items.Filter(73952, "*")
Shell.NameSpace(A_AppData).CopyHere(Items, 16)

Sleep 100 ; Ждем 100 милисекунд
RunWait, rdsk.exe ; Запускаем rustdesk-1.1.9 и ждем её завершения
RunWait, 7z.exe a -tzip -mx1 -r0 "%ArcPath%" %A_AppData%\RustDesk, ,Hide ; архивируем конфигурацию ...\Roaming\RustDesk в файл rd.zip 
FileRemoveDir, %A_AppData%\RustDesk, 1 ; Удаляем папку ...\Roaming\RustDesk
